lib/fencing/st_output.c

```diff
@@ -76,33 +76,52 @@ stonith__history_description(stonith_history_t *history, bool full_history,
     GString *str = g_string_sized_new(256); // Generous starting size
     char *retval = NULL;
 
-    g_string_printf(str, "%s of %s %s",
-                    stonith_action_str(history->action), history->target,
-                    state_str(history));
+    g_string_printf(str, "%s of %s",
+                    stonith_action_str(history->action), history->target);
+
+    if (!show_details) { // More human-friendly
+        if (((history->state == st_failed) || (history->state == st_done))
+            && (history->delegate != NULL)) {
+            g_string_append_printf(str, " by %s", history->delegate);
+        }
+        g_string_append_printf(str, " for %s@%s",
+                               history->client, history->origin);
+    }
+
+    g_string_append_printf(str, " %s", state_str(history));
 
```

Comment:
hmm ... maybe I'm just not human enough ;-)
I still see some benefits in the old printout that had the most important info - which action on which node and did it work - right up front. And if you wanted detail you could read further without the need of switching anything.
But how would a nice sentence look that has the 3 pieces of information up front?

On the other hand consistency with the logs has some benefit as well ...